packages/compass-utils/src/filesystem.ts

```diff
@@ -0,0 +1,145 @@
+import { promises as fs, createWriteStream } from 'fs';
+import path from 'path';
+import { glob } from 'glob';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+import { getAppPath } from './electron';
+
+const { log, mongoLogId } = createLoggerAndTelemetry('COMPASS-UTILS');
+
+type FilesystemOptions<T = unknown> = {
+  subdir: string;
+  onSerialize: (content: T) => string;
+  onDeserialize: (content: string) => T;
+};
+
+export class Filesystem<T> {
```

Comment:
`Filesystem` is probably not a great name for this if it is meant to be used only for files with a specific schema

Reply:
+1